File: iAsync.ObjC/lib/JFFUtils/NSString/NSString+LocalizedPrice.m
#import "NSString+LocalizedPrice.h"

@implementation NSString (LocalizedPrice)

+ (instancetype)localizedPrice:(NSNumber *)price
                   priceLocale:(NSLocale *)priceLocale
{
    NSNumberFormatter *numberFormatter = [NSNumberFormatter new];
    [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
    [numberFormatter setNumberStyle:NSNumberFormatterCurrencyStyle];
    [numberFormatter setLocale:priceLocale];
    NSString *result = [numberFormatter stringFromNumber:price];
    return result;
}

@end

File: iAsync.Swift/lib/JUtils/JUtils/String/NSString+LocalizedPrice.swift
//
//  NSString+LocalizedPrice.swift
//  JUtils
//
//  Created by Vladimir Gorbenko on 09.06.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import Foundation

public extension NSString {
    
    class func localizedPrice(price: NSNumber, priceLocale: NSLocale) -> NSString {
        
        let numberFormatter = NSNumberFormatter()
        
        numberFormatter.formatterBehavior = NSNumberFormatterBehavior.Behavior10_4
        numberFormatter.numberStyle       = NSNumberFormatterStyle.CurrencyStyle
        numberFormatter.locale            = priceLocale
        
        let result = numberFormatter.stringFromNumber(price)
        return result!
    }
}
